src/forms/SpatialBoundsField.tsx

```diff
@@ -0,0 +1,199 @@
+import React, { Component } from "react";
+import { getBoundsFromWmsLayer } from "../utils/getBoundsFromGeoServer";
+import ClearInputButton from "./ClearInputButton";
+import styles from "./DurationField.css";
+import formStyles from "../styles/Forms.css";
+import inputStyles from "../styles/Input.css";
+import thresholdsStyles from './ThresholdsSelection.css';
+
+interface SpatialBoundsProps {
+    value: {
+        north: number,
+        east: number,
+        south: number,
+        west: number
+    } | null,
+    otherValues: {
+        wmsLayerName: string,
+        wmsLayerSlug: string,
+        wmsLayerUrl: string,
+    },
+    valueChanged: Function,
+};
+
+// Validator
+export const spatialBoundsValidator = (fieldValue: SpatialBoundsProps['value']) => {
+    if (fieldValue) {
+        const {
+            north,
+            east,
+            south,
+            west
+        } = fieldValue;
+
+        if (
+            !north ||
+            !east ||
+            !south ||
+            !west
```

Comment:
This is wrong as 0 is a valid value, and "somestring" is not.

I think the check should be on `typeof north === "number"`.

Reply:
I changed the check to Number.isNaN(north) ... because north can only be NaN or a number while typeof NaN also === "number"